### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by MarkVed17**
- Added auth services
- Updated Sign-in screen
- Updated Sign-up screen
- Refactored components wrt auth
- Added Toast for notifications

### CODE REVIEW

File: src/frontend/components/ProfileSection.js [hunk 1]
```diff
@@ -1,23 +1,26 @@
...
 
 const ProfileSection = () => {
+  const { user } = useSelector((state) => state.auth);
+
   return (
     <div className="flex p-2 gap-5">
       <img
         className="w-24 h-24 rounded-full sm:w-28 sm:h-28"
-        src="https://pbs.twimg.com/profile_images/1413581804653617157/Lb6QIsaO_400x400.jpg"
+        src={user.avatar}
```
Comment (viru2001) [inline src/frontend/components/ProfileSection.js:11]: you can destructure `user`

**Review by viru2001 — APPROVED** [review]
code is clean and easily understandable.

**Review by horsemaker — APPROVED** [review]
The code looks clean & structured. Good work here! 👌

**Review by t007rushi — APPROVED** [review]
Good folder structure and naming convention right
Good to merge

File: src/frontend/components/ProfileSection.js (hunk 1 above)
Comment (MarkVed17) [reply src/frontend/components/ProfileSection.js:11]: Yeah, you're right

**Pull request merged**